src/expandable-section/expandable-section-header.tsx

```diff
@@ -141,30 +130,9 @@ const ExpandableContainerHeader = ({
       };
 
   return (
-    <div id={id} className={className} onClick={onClick}>
-      <Wrapper>
-        <span
-          className={isContainer ? styles['header-container-button'] : styles['header-button']}
-          role="button"
-          tabIndex={0}
-          onKeyUp={onKeyUp}
-          onKeyDown={onKeyDown}
-          aria-label={ariaLabel}
-          // Do not use aria-labelledby={id} but ScreenreaderOnly because safari+VO does not read headerText in this case.
-          aria-labelledby={ariaLabel || !isContainer ? undefined : screenreaderContentId}
-          aria-controls={ariaControls}
-          aria-expanded={expanded}
-        >
-          <span className={clsx(styles['icon-container'], styles[`icon-container-${variant}`])}>{icon}</span>
-          <span>{children}</span>
-        </span>
-      </Wrapper>
-      {isContainer && (
-        <ScreenreaderOnly id={screenreaderContentId}>
-          {children} {headerCounter} {headerDescription}
-        </ScreenreaderOnly>
-      )}
-    </div>
+    <ExpandableDefaultHeader {...props} variant={variant}>
```

Comment:
I couldn't find a strong reason for the underlying DOM here to be so different, so have tried to standardize the different variants

Reply:
The reason is to not have heading inside button, see AWSUI-18961

Reply:
this part was also significantly changed in a [recent PR](https://github.com/cloudscape-design/components/pull/1044)

Reply:
ah got it, thanks! will take another look

Reply:
Updated to address this with styling only. That does mean a slight regression in terms of wrapping behavior of the header & icon (see screenshot tests), but it also used to be this way in the past, so I think this is acceptable

Reply:
What about we do not use counter in Header but add right after child? Like 
```
<span className={clsx(styles['icon-container'], styles[`icon-container-${variant}`])}>{icon}</span>
<span>{children} {headerCounter !== undefined && <span className={styles.counter}> {headerCounter}</span>}</span> 
```

Reply:
I'd rather keep as much as possible relying on the <Header> component rather than duplicating things in here